src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -71,38 +78,41 @@ def fetch_group_to_event_data(
     return buffer.backend.get_hash(model=model, field=field)
 
 
-def get_dcg_group_workflow_data(
-    worklow_event_dcg_data: dict[str, str]
-) -> tuple[dict[int, set[int]], dict[int, int], list[Any]]:
+def get_dcg_group_workflow_detector_data(
+    workflow_event_dcg_data: dict[str, str]
+) -> tuple[DataConditionGroupGroups, DataConditionGroupWorkflow, DataConditionGroupDetector]:
     """
-    Parse the data in the buffer hash, which is in the form of {workflow_id}:{event_id}:{dcg_id, ..., dcg_id}:{dcg_type}
+    Parse the data in the buffer hash, which is in the form of {workflow/detector_id}:{event_id}:{dcg_id, ..., dcg_id}:{dcg_type}
     """
 
-    dcg_to_groups: dict[int, set[int]] = defaultdict(set)
-    dcg_to_workflow: dict[int, int] = {}
-    all_event_data = []
+    dcg_to_groups: DataConditionGroupGroups = defaultdict(set)
+    dcg_to_workflow: DataConditionGroupWorkflow = {}
+    dcg_to_detector: DataConditionGroupDetector = {}
```

Comment:
Could we treat these a bit more generically? like we have a list of DataCondition types already (enum); we could use that to say: `dcg_to_handlers: dict[HandlerType, set[handlers]]`

That would allow us to dynamically drive the number of handlers from the number of handlers we have defined. what's super nice about that is we could add more hooks into the system or remove hooks w/o having to update this logic code.

Reply:
ok discussed and we want a nested dict indexed by `DataConditionHandlerType` with the same existing `dict[int,int]` inside each one